sdk/eventgrid/eventgrid/src/sharedAccessSignitureCredential.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Represents a credential defined by a shared signature.
+ */
+export interface SignatureCredential {
+  /**
+   * The signature to be used in authentication.
+   */
+  signature: string;
+}
+
+/**
+ * A credential type which allows using a "shared access signature" to authenticate with EventGrid
+ * when sending events.
+ *
+ * A shared access signiture may be generated by using `EventGridClient.generateSharedAccessSignature`
+ */
+export class EventGridSharedAccessSignatureCredential implements SignatureCredential {
+  private _signature: string;
+
+  /**
+   * The value of the signature to be used in authentication
+   */
+  public get signature(): string {
```

Comment:
You should avoid using getters if possible, since it means you have to downlevel your public TS interfaces with `downlevel-dts` to avoid a breaking change made in TS 3.7.

Reply:
I was sort of trying to mirror the design of `AzureKeyCredential` and `KeyCredential`, which uses a getter as well, on the actual class (ref: [core-auth.api.md](https://github.com/Azure/azure-sdk-for-js/blob/%40azure/core-auth_1.1.3/sdk/core/core-auth/review/core-auth.api.md)).

I would be happy to not follow that pattern and instead do: 

```typescript
export interface SignatureCredential {
  signature(): string;
}

export class EventGridSharedAccessSignatureCredential implements SignatureCredential {
  constructor(signature: string);
  public signature(): string;
  public update(newSignature: string): void;
}
```

If that's preferable.  I don't know enough to understand if there was some reason that the existing `KeyCredential` and `AzureKeyCredential` types were designed the way they are.

Thoughts?

Reply:
I only dimly remember the discussion for why we went with a getter; I do remember complaining about it since it caused the same trouble. @willmtemple do you remember the details?

Reply:
FWIW, I've moved to the above pattern that I outlined above.  Happy to move back (and downlevel if needed) if @willmtemple would prefer that.

Reply:
(Resolving this conversation, but let me know if you want me to change anything, @willmtemple)